### DISCUSSION TITLE
Why does .container use --bs-gutter-width for padding?

### DISCUSSION THREAD

**Discussion by donquixote** [twbs/bootstrap#35807] [Q&A]
The .container has this rule:

```css
.container, .container-fluid, .container-lg, .container-md, .container-sm, .container-xl, .container-xxl {​​​​​​​​​​​​​​
   padding-left: var(--bs-gutter-x,.75rem);
   padding-right: var(--bs-gutter-x,.75rem);
```

However, typically at this point, the `--bs-gutter-width` variable is not defined, so it will fall back to `.75rem`.

Then in .row we have this:

```css
.row {
  // This value depends on a SASS variable, I think. The front-end people set this to 1.5rem for a project I work on.
  --bs-gutter-x: 1.5rem;
  margin-left: calc(var(--bs-gutter-x)*-.5);
  margin-right: calc(var(--bs-gutter-x)*-.5);
```

This is meant to compensate the padding on columns:
```css
.row > * {
    padding-left: calc(var(--bs-gutter-x)*.5);
    padding-right: calc(var(--bs-gutter-x)*.5);
```

If you now have a .container within a .row, e.g. for a nested full-width section, the padding will be off:
https://codepen.io/donqi/pen/BamWyRZ

```html
<div class="container">
  <h2>Starting the container</h2>
  <div class="row">
    <div class="col-*">
      <h2>Starting a column</h3>
      <div class="full-width" style="background-image: ...">
        <div class="container">
          <h2>Container inside a full-width banner</h4>
```

So:
- If .container is outside any .row, then using the `--bs-gutter-x` variable has no effect.
- If .container is inside full-width section inside a .row, then using the `--bs-gutter-x` variable will cause mismatch with outer containers.
- If a .container is used in any other place, that's probably wrong :)

**Comment by florianlacreuse** [comment]
Related to #35282?

**Reply by donquixote** [reply]
Related, yes. Duplicate, not sure.
I think it can be worthwhile to have this separate Q/A to document the overall design choice, instead of adding to the other issue.

The symptom described in #35282 is different from my own issue.
But it reminds me I should craft a codepen!

**Reply by donquixote** [reply]
Codepen added to the question above. Here it is again, https://codepen.io/donqi/pen/BamWyRZ
